packages/react-core/src/components/Page/PageHeaderToolsItem.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Page/page';
+import { css } from '@patternfly/react-styles';
+
+export interface PageHeaderToolsItemProps extends React.HTMLProps<HTMLDivElement> {
+  /** Content rendered in page header tools item */
+  children: React.ReactNode;
+  /** Makes the group only visible for desktop sizes (>= lg breakpoint) */
+  visibleOnLg?: boolean;
+  /** Hides the group for desktop sizes (< lg breakpoint) */
+  hiddenOnLg?: boolean;
+  /** Hides the group for mobile sizes (< md breakpoint). Typically used on a user dropdown */
+  hiddenOnSm?: boolean;
+  /** Can apply the selected modifier to a button */
+  isSelected?: boolean;
+}
+
+/**
+ * This component wraps the child and applies styling such as selected on the thing passed in, example:
+ *
+ * <PageHeaderToolsItem isSelected><Button isPlain>Hello</Button></PageHeaderToolsItem>
+ * will render
+ * <button class="pf-m-plain pf-m-selected">Hello</button>
+ */
+export const PageHeaderToolsItem: React.FunctionComponent<PageHeaderToolsItemProps> = ({
```

Comment:
I'm curious how this component works. I think it would be more consistent if we had a `.pf-c-page__header-tools-item` class that you could apply `hidden/visible` to. I'm assuming that would still allow us to apply `.pf-m-selected` to a `<Button>` used in `.pf-c-page__header-tools-item` if it has the `isSelected` prop?

Reply:
So the problem is, in HTML you can just add the `pf-m-selected` class to a `button` anywhere. If you wanted to do this in react you'd have to manually add this class name yourself i.e. 
```
<PageHeaderTools>
  <PageHeaderToolsGroup>
      <Button className="pf-m-selected">
          <CogIcon />
      </Button>
  </PageHeaderToolsGroup>
</PageHeaderTools>
```
because the selected modifier is not a prop we have available for Buttons, the class is defined in the context of page.

Wrapping the `<Button>` component in `<PageHeaderToolsItem>` basically allows me to pass the `pf-m-selected` class or any other potential modifier down to the child component.